### ISSUE TITLE
docker-machine not authenticating correctly with generic driver

### ISSUE THREAD

**Issue by tino** [docker/machine#2400]
I am trying to add an existing remote server with the generic driver on OS X El Capitan, with the latest Toolbox installed (1.9.1). Command:

`docker-machine create -d generic --generic-ssh-key=~/.ssh/id_no_password --generic-ip-address=xxx.x.xxx.xxx --generic-ssh-user=tino my-generic-machine`

This works, and `docker-machine ls` shows the machine as running. However, `docker-machine env my-generic-machine` fails with:

```
Error running connection boilerplate: Error checking and/or regenerating the certs: There was an error validating certificates for host "xxx.x.xxx.xxx:2376": dial tcp xxx.x.xxx.xxx:2376: i/o timeout
You can attempt to regenerate them using 'docker-machine regenerate-certs name'.
Be advised that this will trigger a Docker daemon restart which will stop running containers.
```

Running `docker-machine regenerate-certs my-generic-machine` works, and all three certificates are the same on both sides when I manually compare them.

On the server side, in `/var/log/auth.log` I see these lines when docker-machine commands fail:

`sshd[24702]: Did not receive identification string from xxx.xx.xxx.xx`

Why is docker-machine not authenticating correctly? Or is there something on the server not installed correctly?

When I manually set the `DOCKER_HOST`, `DOCKER_CERT_PATH`,  `DOCKER_MACHINE_NAME` and `DOCKER_DRIVER=generic` env vars to the values for the remote machine, `docker ps` just says (after a long time):

`Cannot connect to the Docker daemon. Is the docker daemon running on this host?`

Then the authentication error is not shown in the logs.

Versions:
```
# docker-machine --version (on my Mac)
docker-machine version 0.5.1 (7e8e38e)
# docker --version  (on both Mac and server)
Docker version 1.9.1, build a34a1d5
```

**Comment by tino** [comment]
Btw, I get the same results from an Ubuntu 14.04.1 LTS as client machine

**Comment by driesdesmet** [comment]
I'm getting the same. "docker-machine ssh [machinename]" works, I can log in, but I noticed a "Did not receive identification string from..." when I exit the session.

"docker-machine ls" doesn't authenticate properly.

**Comment by robertjung** [comment]
Same here. Breaks on `sudo apt-get update`:

```
...
(prod1) Calling RPCServerDriver.GetSSHHostname
(prod1) Calling RPCServerDriver.GetSSHPort
(prod1) Calling RPCServerDriver.GetSSHKeyPath
(prod1) Calling RPCServerDriver.GetSSHUsername
Using SSH client type: external
About to run SSH command:
if grep -xq 127.0.1.1.* /etc/hosts; then sudo sed -i 's/^127.0.1.1.*/127.0.1.1 prod1/g' /etc/hosts; else echo '127.0.1.1 prod1' | sudo tee -a /etc/hosts; fi
SSH cmd err, output: <nil>:
(prod1) Calling RPCServerDriver.GetSSHHostname
(prod1) Calling RPCServerDriver.GetSSHPort
(prod1) Calling RPCServerDriver.GetSSHKeyPath
(prod1) Calling RPCServerDriver.GetSSHUsername
Using SSH client type: external
About to run SSH command:
sudo apt-get update
SSH cmd err, output: exit status 255:
Error creating machine: Error running provisioning: Something went wrong running an SSH command!
command : sudo apt-get update
err     : exit status 255
output  :
```
with docker-machine 0.5 and 0.5.1

**Comment by jeanlaurent** [comment]
@tino , @driesdesmet and @robertjung Can you report your target ( server ) OS + version ?

**Comment by robertjung** [comment]
Virgin Ubuntu 14.04, authenticated by public key as root:

```
docker-machine --debug create -d generic \
--generic-ssh-user root \
--generic-ip-address IP \
name
```

**Comment by gmlove** [comment]
I've encountered the same problem.

My docker host os:
```
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.3 LTS
Release:	14.04
Codename:	trusty
```

client os: OS X EI Version 10.11.1

**Comment by jeanlaurent** [comment]
Strange, got it working 1st time on a `Ubuntu 14.04.3 LTS`. This is not the provisioner then, this is something in between, most likely the certs.

Can you all can ssh into your machine on the same client host system ?

**Comment by robertjung** [comment]
yes, `ssh root@IP` works just fine. the first steps of the setup run just fine also. docker-machine managed to drop of the line `127.0.1.1 machinename` in `/etc/hosts` for example

**Comment by robertjung** [comment]
Found the problem. `ufw limit ssh` doesn't play nicely with provisioners that create new ssh connections for each action, it consistently blocked the 6th request in a 30 second interval that happened to be the `apt-get` command *facepalm*

Sorry for adding to the confusion!

**Comment by jeanlaurent** [comment]
Hey Thanks for your input @robertjung , this helps !

**Comment by tino** [comment]
@jeanlaurent Target machine runs a virgin Ubuntu 14.04.3 LTS (GNU/Linux 3.13.0-68-generic x86_64). ssh'in in works normally. That is also what I used to compare the certificates.

In my case, `ufw` is inactive so that should not be the culprit.

**Comment by driesdesmet** [comment]
My server target is a GCE machine:
dries@cloud:~$ cat /etc/lsb-release 

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.3 LTS"
```

I can also do a few manual ssh logins to this machine, but with  `docker-machine ls` the running sshguard blocks me out.

**Comment by gmlove** [comment]
I think I found the answer, please goto http://docs.mist.io/article/20-adding-docker.
I've already test the first method. It works.
And I use `openssl x509 -in ca.pem -text` to show the ca content on the docker container, no  “Common Name” found.
Seems the root cause is the way that docker-machine generates file ca.pem.

**Comment by andrewbranch** [comment]
This was pretty dumb of me to not realize for several hours, but I forgot to open up port 2376 to inbound traffic on my Azure VM. If you’re getting this error... well, there’s actually a chance that it’s exactly what it sounds like.
